src/System.Security.Claims/src/System/Security/Claims/Claim.cs

```diff
@@ -0,0 +1,550 @@
+        /// </summary>
+        public virtual Claim Clone()
+        {
+            return Clone((ClaimsIdentity)null);
+        }
+
+        /// <summary>
+        /// Creates a new instance <see cref="Claim"/> with values copied from this object.
+        /// </summary>
+        /// <param name="identity">the value for <see cref="Claim.Subject"/>, which is the <see cref="ClaimsIdentity"/> that has these claims.
+        /// <remarks><see cref="Claim.Subject"/> will be set to 'identity'.</remarks>
+        public virtual Claim Clone(ClaimsIdentity identity)
+        {
+            return new Claim(this, identity);
+        }
+
+        private void Initialize(BinaryReader reader, ClaimsIdentity subject)
+        {
+            if (reader == null)
+            {
+                throw new ArgumentNullException("reader");
+            }
+
+            _subject = subject;
+
+            SerializationMask mask = (SerializationMask)reader.ReadInt32();
+            int numPropertiesRead = 1;
+            int numPropertiesToRead = reader.ReadInt32();
+            _value = reader.ReadString();
+
+            if ((mask & SerializationMask.NameClaimType) == SerializationMask.NameClaimType)
+            {
+                _type = ClaimsIdentity.DefaultNameClaimType;
+            }
+            else if ((mask & SerializationMask.RoleClaimType) == SerializationMask.RoleClaimType)
+            {
+                _type = ClaimsIdentity.DefaultRoleClaimType;
+            }
+            else
+            {
+                _type = reader.ReadString();
+                numPropertiesRead++;
+            }
+
+            if ((mask & SerializationMask.StringType) == SerializationMask.StringType)
+            {
+                _valueType = reader.ReadString();
+                numPropertiesRead++;
+            }
+            else
+            {
+                _valueType = ClaimValueTypes.String;
+            }
+
+            if ((mask & SerializationMask.Issuer) == SerializationMask.Issuer)
+            {
+                _issuer = reader.ReadString();
+                numPropertiesRead++;
+            }
+            else
+            {
+                _issuer = ClaimsIdentity.DefaultIssuer;
+            }
+
+            if ((mask & SerializationMask.OriginalIssuerEqualsIssuer) == SerializationMask.OriginalIssuerEqualsIssuer)
+            {
+                _originalIssuer = _issuer;
+            }
+            else if ((mask & SerializationMask.OriginalIssuer) == SerializationMask.OriginalIssuer)
+            {
+                _originalIssuer = reader.ReadString();
+                numPropertiesRead++;
+            }
+            else
+            {
+                _originalIssuer = ClaimsIdentity.DefaultIssuer;
+            }
+
+            if ((mask & SerializationMask.HasProperties) == SerializationMask.HasProperties)
+            {
+                // TODO - brentsch, maximum # of properties
```

Comment:
Is there an associated GitHub issue for this or other TODOs?  It'd be good to include the issue # here.